packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -426,6 +430,15 @@ export class StateMachine extends StateMachineBase {
     this.role.addToPrincipalPolicy(statement);
   }
 
+  private validateStateMachineName(stateMachineName: string) {
+    if (stateMachineName.length < 1 || stateMachineName.length > 80) {
+      throw new Error('StateMachine name length must be between 1 and 80 characters');
```

Comment:
also provide the input that was supplied when throwing this error so users have a better idea around why validation failed.

```suggestion
      throw new Error(`State Machine name must be between 1 and 80 characters. received: ${stateMachineName}`);
```